lib/entry/StateManager.js

```diff
@@ -131,26 +126,21 @@ module.exports = function(liveEntry) {
     }
 
     function updateStreamInfo(currentTime) {
-        if ((currentTime - lastStreamInfoUpdateTime) >= streamInfoUpdateInterval) {
-            return getEntryServerNodeId()
-                .then(function () {
-                    if (liveEntry.isMultipleAudio) {
-                        logger.debug("Stream is Multiple Audio. Skipping server update");
-                        lastStreamInfoUpdateTime = new Date();
-                        return Q.resolve();
-                    }
-                    else {
-                        return updateFlavorsToServer(liveEntry.entryId, liveEntry.entryServerNodeId, liveEntry.flavorsDownloaders);
-                    }
-                })
-                .catch(function (err) {
-                    logger.warn("entryServerNodeId not found, stream info will not update! Error: %s", ErrorUtils.error2string(err));
-                    liveEntry.entryServerNodeId = null;
-                    return Q.reject(err);
-                });
+        let timeDiff = Math.abs(currentTime.getTime() - lastStreamInfoUpdateTime.getTime());
+
+        if (+timeDiff >= +streamInfoUpdateInterval) {
+            if (liveEntry.isMultipleAudio) {
+                logger.debug("Stream is Multiple Audio. Skipping server update");
+                lastStreamInfoUpdateTime = new Date();
+                return Q.resolve();
+            }
+            else {
+                return updateFlavorsToServer(liveEntry.entryId, liveEntry.entryServerNodeId, liveEntry.flavorsDownloaders);
+            }
```

Comment:
you removed the   liveEntry.entryServerNodeId = null;		 in case of a catch.

this was added by Yossi on purpose... don't remember the bug that it fixed, please added it again and talk with him

Reply:
code moved back though it was done due to entryServerNode being taken from entryInfo object received before call to registerEntryInDatabase... so EntryServerNodeId is undefined (null)